snapshots/windows/windows.go

```diff
@@ -193,17 +197,72 @@ func (s *snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 	// grab the existing id
 	id, _, _, err := storage.GetInfo(ctx, key)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to get storage info for %s", key)
 	}
 
-	usage, err := fs.DiskUsage(ctx, s.getSnapshotDir(id))
+	snapshot, err := storage.GetSnapshot(ctx, key)
 	if err != nil {
 		return err
 	}
 
+	path := s.getSnapshotDir(id)
+
+	// In order to turn a Scratch Layer (sandbox.vhdx with all the data) into a
+	// Read-only Layer (Files/, Hives/, etc) we need to create a new snapshot and
+	// restream the data out of the old snapshot.
+
+	// Some Snapshots will already have Files/ etc, if they were created by streaming in a layer
+	// archive. Layers created during image pull will be in this state, for example. In that case,
+	// we should delete sandbox.vhdx if still present, as it will not reflect the layer content.
+	// A layer with an empty parent list will never have had a sandbox.vhdx.
```

Comment:
TODO: Look at #4643 and see if it affects the logic here or in #4419. I was already handling this by deleting the sandbox.vhdx for imported layers (see the final `else if` clause below), and now it'll never be created, so I can drop that logic completely.

And it seems to have added a formal way of recognising a "for-import" active snapshot by key-name, if I need that here.